cfme/web_ui/__init__.py

```diff
@@ -1650,6 +1650,260 @@ def _fill_radio(radio, value):
     radio.observer_wait(value)
 
 
+class BootstrapTreeview(object):
+    """A class representing the Bootstrap treeview used in newer builds.
+
+    Implements ``expand_path``, ``click_path``, ``read_contents``. All are implemented in manner
+    very similar to the original :py:class:`Tree`.
+
+    Args:
+        tree_id: Id of the tree, the closest div to the root ``ul`` element.
+    """
+    ROOT_ITEM = './ul/li[1]'
+    SELECTED_ITEM = './ul/li[contains(@class, "node-selected")]'
+    CHILD_ITEMS = (
+        './ul/li[starts-with(@data-nodeid, {id}) and not(@data-nodeid={id})'
+        ' and count(./span[contains(@class, "indent")])={indent}]')
+    CHILD_ITEMS_TEXT = (
+        './ul/li[starts-with(@data-nodeid, {id}) and not(@data-nodeid={id})'
+        ' and contains(normalize-space(text()), {text})'
+        ' and count(./span[contains(@class, "indent")])={indent}]')
```

Comment:
Yes, we count indents, because why not. All the tree items are located under one ul and you can detect the parent/child relationship only by that the child's nodeid starts with parent's nodeid and the number of indent tags in the child is number of indent tags in parent + 1 :tired_face: